addons/CLib/MapGraphics/fn_mapGraphicsPosition.sqf

```diff
@@ -23,16 +23,28 @@ params [
     ["_map", controlNull, [controlNull]]
 ];
 
-if (_position isEqualType [] && {(_position select 1) isEqualType []}) then {
-    _position params ["_pos", "_offset"];
+if (_position isEqualType []) then {
+    if ((_position select 1) isEqualType []) then {
+        _position params ["_pos", "_offset"];
+
+        if (_pos isEqualType objNull) then {
+            _pos = getPosVisual _pos;
+        };
+        _pos = _map ctrlMapWorldToScreen _pos;
+        _pos = [(_pos select 0) + (_offset select 0) / 640, (_pos select 1) + (_offset select 1) / 480];
+        _pos = _map ctrlMapScreenToWorld _pos;
+        _position = _pos;
+    };
 
-    if (_pos isEqualType objNull) then {
-        _pos = getPosVisual _pos;
+    // Make sure the returned position is Position3D
+    private _originalLength = count _position;
+    _position resize 3;
+    
+    // If the original position did contain fewer than 3 entries, replace the nil values inserted by the resize commands
+    // with zeros
+    for "_i" from _originalLength to 3 do {
```

Comment:
this is way to complicated. and i think it can be simplified to this
```sqf
if (count _position != 3) then {
   _position resize 3;
   _position set [2, 0];
};
```

Reply:
or we could check seperatly of it bigger than 3 and than resize it. and if is smaller just pushback a 0. if the users adds in anything different than a 2 or 3 vector it is a users fault!